app/controllers/users_controller.rb

```diff
@@ -0,0 +1,23 @@
+class UsersController < ApplicationController
+  respond_to :json
+
+  def show
+    @user = User.find(params[:id])
```

Comment:
find for incorrect id throw Error, use find_by
if it nil handle with correct error message.

repeated @user, make it with before_action